torch/utils/data/standard_pipes.ipynb

```diff
@@ -0,0 +1,1006 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "0\n2\n4\n6\n8\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).filter(lambda x: x % 2 == 0)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "Classic `filter` applies filter function to every element inside mini-batches by default\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[0, 2]\n[4]\n[6, 8]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10)\n",
+    "dp = dp.batch(3).filter(lambda x: x % 2 == 0)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "By default if mini-batch ends with zero elements after filtering it is dropped from response"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[5]\n[6, 7, 8]\n[9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10)\n",
+    "dp = dp.batch(3).filter(lambda x: x > 4)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "You can override this behaviour using `drop_empty_batches` argument"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[]\n[5]\n[6, 7, 8]\n[9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10)\n",
+    "dp = dp.batch(3).filter(lambda x: x > 4, drop_empty_batches=False)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "You can also apply filtering function on top of entire mini-batch using `batch_level` agrument"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10)\n",
+    "dp = dp.batch(3).filter(lambda l: len(l) < 3, batch_level=True)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## Shuffle\n",
+    "\n",
+    "Function: `shuffle`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    "  - `batch_level: bool = False` whether elements from the source DataPipe counted as independant objects or we process mini-batched elements as whole.\n",
+    "  - `buffer_size: int = 10000`\n",
+    "  - `cross_shuffle: bool = True`\n",
```

Comment:
I am thinking about `cross_batch` or `cross_batch_shuffle` as the parameter name ;) 

Also this will error if the originally data is not batched right ? 😃